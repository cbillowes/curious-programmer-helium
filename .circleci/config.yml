version: 2
jobs:
  build:
    docker:
      - image: circleci/node:chakracore-8-browsers-legacy
    steps:
      - checkout
      - run: echo "Chopping onions without the tears"
      - run: npm run build
  test:
    docker:
      - image: circleci/node:chakracore-8-browsers-legacy
    steps:
      - checkout
      - run: echo "Making soup and drinking beers"
      - run: sudo npm install -g jest-cli
      - run: npm run test
  page_speed:
    docker:
      - image: circleci/node:chakracore-8-browsers-legacy
    steps:
      - checkout
      - run: echo "Making sure it won't take years"
      - run: echo "* Deploy to Netlify"
      - run: echo "* Execute lighthouse page speed tests on build"
workflows:
  version: 2
  integration:
    jobs:
      - build
      - test:
          requires:
            - build
      - page_speed:
          requires:
            - test
