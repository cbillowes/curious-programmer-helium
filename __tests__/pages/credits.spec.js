import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"
import Helmet from "react-helmet"
import CreditsPage from "../../src/pages/credits"
import config from "../../data/SiteConfig"

describe("a credits page", () => {
  it("should render", () => {
    const component = <CreditsPage />
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should generate the meta data title", () => {
    const component = mount(<CreditsPage />)
    const helmet = Helmet.peek()
    expect(helmet.title).toEqual(`Credits | ${config.siteTitle}`)
  })

  it("should render children tags", () => {
    const component = mount(<CreditsPage />)
    expect(component.find(".credits-container").html()).toBeDefined()
  })
})