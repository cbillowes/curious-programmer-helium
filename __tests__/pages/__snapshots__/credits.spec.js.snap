// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a credits page should render 1`] = `
<div>
  <div
    className="navigation"
  >
    <div
      className="logo"
    >
      <div
        className="default-logo"
      >
        <a
          aria-current="page"
          className=""
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          style={Object {}}
        >
          { 
          <span
            className="normal-weight"
          >
            curious
          </span>
           
          programmer
           }
        </a>
      </div>
      <div
        className="mobile-logo"
      >
        <a
          aria-current="page"
          className=""
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          style={Object {}}
        >
          { }
        </a>
      </div>
    </div>
    <div
      className="icons"
    >
      <div
        className="icon search "
        onClick={[Function]}
      >
        <i />
      </div>
      <div
        className="icon menu "
        onClick={[Function]}
      >
        <i />
      </div>
    </div>
    <nav
      className="menubar "
      onClick={[Function]}
    >
      <ul>
        <li
          className="menu-item selected"
        >
          <a
            aria-current="page"
            className=""
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={Object {}}
          >
            Blog
          </a>
        </li>
        <li
          className="menu-item "
        >
          <a
            href="/about"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            About
          </a>
        </li>
        <li
          className="menu-item "
        >
          <a
            href="/tags"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Tags
          </a>
        </li>
        <li
          className="menu-item "
        >
          <a
            href="/credits"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Credits
          </a>
        </li>
        <li
          className="menu-item "
        >
          <a
            href="/privacy-policy"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Privacy Policy
          </a>
        </li>
        <li
          className="menu-item "
        >
          <a
            href="/license"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            License
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="searchbar "
    >
      <input
        className="search-query"
        onKeyUp={[Function]}
        type="text"
      />
      <div
        className="results"
      >
        <ul>
          <li
            className="empty"
          >
            What do you feel like reading today? Search as you type, also, enter works too.
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="credits-container container"
  >
    <h1>
      Credits
    </h1>
    <section>
      <h1>
        Hosting
      </h1>
      <div
        className="credit"
      >
        <span>
          My high performance cloud server which I use to power my Solr search.
        </span>
        <a
          className=""
          href="https://www.cloudafrica.net/"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          CloudAfrica
        </a>
      </div>
      <div
        className="credit"
      >
        <span>
          An all-in-one workflow that combines global deployment, continuous integration, and automatic HTTPS.
        </span>
        <a
          className=""
          href="https://netlify.com"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          Netlify
        </a>
      </div>
    </section>
    <section>
      <h1>
        Continuous Integration
      </h1>
      <div
        className="credit"
      >
        <span>
          Travis CI is a hosted, distributed continuous integration service used to build and test software projects hosted at GitHub.
        </span>
        <a
          className=""
          href="https://travis-ci.org"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          Travis CI
        </a>
      </div>
      <div
        className="credit"
      >
        <span>
          Get automated code review for test coverage, complexity, duplication, security, style, and more.
        </span>
        <a
          className=""
          href="https://codeclimate.com"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          Code Climate
        </a>
      </div>
    </section>
    <section>
      <h1>
        Code
      </h1>
      <div
        className="credit"
      >
        <span>
          Gatsby lets you build blazing fast sites with your data, whatever the source. Liberate your sites from legacy CMSs and fly into the future.
        </span>
        <a
          className=""
          href="https://www.gatsbyjs.org/"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          gatsbyjs.org
        </a>
      </div>
      <div
        className="credit"
      >
        <span>
          A high performance skeleton starter for GatsbyJS that focuses on SEO/Social features/development environment.
        </span>
        <a
          className=""
          href="https://github.com/Vagr9K/gatsby-advanced-starter"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          Vagr9K/gatsby-advanced-starter
        </a>
      </div>
      <div
        className="credit"
      >
        <span>
          Adds syntax highlighting to code blocks in markdown files using PrismJS.
        </span>
        <a
          className=""
          href="https://www.gatsbyjs.org/packages/gatsby-remark-prismjs/"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          gatsby-remark-prismjs
        </a>
      </div>
      <div
        className="credit"
      >
        <span>
          Embed Gist snippets in markdown.
        </span>
        <a
          className=""
          href="https://www.gatsbyjs.org/packages/gatsby-remark-embed-gist/"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          gatsby-remark-embed-gist
        </a>
      </div>
      <div
        className="credit"
      >
        <span>
          Embed a Youtube Videos in markdown.
        </span>
        <a
          className=""
          href="https://www.gatsbyjs.org/packages/gatsby-remark-embed-youtube/"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          gatsby-remark-embed-youtube
        </a>
      </div>
    </section>
    <section>
      <h1>
        Icons
      </h1>
      <div
        className="credit"
      >
        <span>
          Music Searcher free icon
        </span>
        <a
          className=""
          href="https://www.flaticon.com/free-icon/musica-searcher_70376"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          Egor Rumyantsev
        </a>
      </div>
      <div
        className="credit"
      >
        <span>
          Menu free icon
        </span>
        <a
          className=""
          href="https://www.flaticon.com/free-icon/menu_483345"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          Those Icons
        </a>
      </div>
      <div
        className="credit"
      >
        <span>
          Github free icon
        </span>
        <a
          className=""
          href="https://www.flaticon.com/free-icon/github_1051275"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          Freepik
        </a>
      </div>
      <div
        className="credit"
      >
        <span>
          Twitter free icon
        </span>
        <a
          className=""
          href="https://www.flaticon.com/free-icon/twitter_185961"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          Freepik
        </a>
      </div>
      <div
        className="credit"
      >
        <span>
          LinkedIn free icon
        </span>
        <a
          className=""
          href="https://www.flaticon.com/free-icon/linkedin_185964"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          Freepik
        </a>
      </div>
      <div
        className="credit"
      >
        <span>
          Gmail free icon
        </span>
        <a
          className=""
          href="https://www.flaticon.com/free-icon/gmail_270021"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          Freepik
        </a>
      </div>
      <div
        className="credit"
      >
        <span>
          Rss SVG Vector
        </span>
        <a
          className=""
          href="https://www.svgrepo.com/svg/95294/rss"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          SVG Repo
        </a>
      </div>
      <div
        className="credit"
      >
        <span>
          Stackoverflow icon based on 
          <a
            className=""
            href="https://www.iconfinder.com/iconsets/miu-gloss-social"
            rel="nofollow noopener noreferrer "
            target="_blank"
          >
            Miu Gloss Social
          </a>
           set
        </span>
        <a
          className=""
          href="https://www.iconfinder.com/icons/287623/stackoverflow_icon"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          Icon Finder
        </a>
      </div>
      <div>
        Icons made by 
        <a
          href="https://www.flaticon.com/authors/those-icons"
          title="Those Icons"
        >
          Those Icons
        </a>
         from 
        <a
          href="https://www.flaticon.com/"
          title="Flaticon"
        >
          www.flaticon.com
        </a>
         is licensed by 
        <a
          href="http://creativecommons.org/licenses/by/3.0/"
          target="_blank"
          title="Creative Commons BY 3.0"
        >
          CC 3.0 BY
        </a>
        .
      </div>
      <div>
        Icons made by 
        <a
          href="https://www.flaticon.com/authors/egor-rumyantsev"
          title="Egor Rumyantsev"
        >
          Egor Rumyantsev
        </a>
         from 
        <a
          href="https://www.flaticon.com/"
          title="Flaticon"
        >
          www.flaticon.com
        </a>
         is licensed by 
        <a
          href="http://creativecommons.org/licenses/by/3.0/"
          target="_blank"
          title="Creative Commons BY 3.0"
        >
          CC 3.0 BY
        </a>
        .
      </div>
      <div>
        Icons made by 
        <a
          href="https://www.freepik.com/"
          title="Freepik"
        >
          Freepik
        </a>
         from 
        <a
          href="https://www.flaticon.com/"
          title="Flaticon"
        >
          www.flaticon.com
        </a>
         is licensed by 
        <a
          href="http://creativecommons.org/licenses/by/3.0/"
          target="_blank"
          title="Creative Commons BY 3.0"
        >
          CC 3.0 BY
        </a>
        .
      </div>
    </section>
    <section>
      <h1>
        Photography
      </h1>
      <div
        className="credit"
      >
        <span>
          Lorem Ipsum... but for photos.
        </span>
        <a
          className=""
          href="https://picsum.photos"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          picsum.photos
        </a>
      </div>
    </section>
    <section>
      <h1>
        Tools
      </h1>
      <div
        className="credit"
      >
        <span>
          Visual Studio Code is an open-source and free source code editor developed by Microsoft for Windows, Linux and macOS. It includes support for debugging, embedded Git control, syntax highlighting, intelligent code completion, snippets, and code refactoring.
        </span>
        <a
          className=""
          href="https://code.visualstudio.com/"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          Microsoft Visual Studio Code
        </a>
      </div>
      <div
        className="credit"
      >
        <span>
          Pixlr is a cloud-based set of image tools and utilities, including a number of photo editors, a screen recorder browser extension, and a photo sharing service. The suite was intended for non-professionals, however the apps range from simple to advanced photo editing.
        </span>
        <a
          className=""
          href="https://pixlr.com/editor/"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          pixlr.com
        </a>
      </div>
      <div
        className="credit"
      >
        <span>
          Convert your images to Base64.
        </span>
        <a
          className=""
          href="https://www.base64-image.de/"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          Base64 Image
        </a>
      </div>
      <div
        className="credit"
      >
        <span>
          SVG editor: Create unique artwork with shapes, color gradients and layers. Use Vector Paint for posters, web graphics or diagrams.
        </span>
        <a
          className=""
          href="http://vectorpaint.yaks.co.nz/"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          Vector Paint
        </a>
      </div>
      <div
        className="credit"
      >
        <span>
          SVG editor: Edit and save images.
        </span>
        <a
          className=""
          href="https://editor.method.ac"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          Method draw
        </a>
      </div>
    </section>
    <section>
      <h1>
        Privacy Policy
      </h1>
      <div
        className="credit"
      >
        <span>
          Attorney-level solutions to make your websites and apps compliant with the law across multiple countries and legislations.
        </span>
        <a
          className=""
          href="https://www.iubenda.com"
          rel="nofollow noopener noreferrer "
          target="_blank"
        >
          iubenda
        </a>
      </div>
    </section>
  </div>
  <div
    className="footer-container not-for-print"
  >
    <footer
      className="footer"
    >
      <ul
        className="social-links-footer"
      >
        <li
          className="link"
        >
          <a
            className="github"
            href="https://github.com/cbillowes"
            rel="nofollow noopener noreferrer "
            target="_blank"
          />
        </li>
        <li
          className="link"
        >
          <a
            className="twitter"
            href="https://twitter.com/cbillowes"
            rel="nofollow noopener noreferrer "
            target="_blank"
          />
        </li>
        <li
          className="link"
        >
          <a
            className="linkedin"
            href="https://www.linkedin.com/in/cbouwer/"
            rel="nofollow noopener noreferrer "
            target="_blank"
          />
        </li>
      </ul>
      <div
        className="copyright-footer"
      >
        Copyright © 
        
        2019
         
        
        Curious Programmer
        . 
        
        Made with Gatsby v2 and other 
        
        <a
          href="/credits"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          cool stuff
        </a>
        .
      </div>
      <div
        className="license-footer"
      >
        <span
          className="license"
        >
          <a
            className="license-icon"
            href="http://creativecommons.org/licenses/by-sa/4.0/"
            rel="nofollow noopener noreferrer license"
            target="_blank"
          >
            <img
              alt="Creative Commons License"
              src="https://i.creativecommons.org/l/by-sa/4.0/88x31.png"
              title="Creative Commons Attribution-ShareAlike 4.0 International License"
            />
          </a>
          <p>
            <span
              property="dct:title"
              xmlns:dct="http://purl.org/dc/terms/"
            >
               
              Curious Programmer
               content
               
            </span>
             
            by
             
            <a
              className=""
              href="https://curiousprogrammer.io"
              rel="nofollow noopener noreferrer cc:attributionURL"
              target="_blank"
            >
              Clarice Bouwer
            </a>
             
            is licensed under a
             
            <a
              className=""
              href="http://creativecommons.org/licenses/by-sa/4.0/"
              rel="nofollow noopener noreferrer license"
              target="_blank"
            >
              Creative Commons Attribution-ShareAlike 4.0 International License
            </a>
            .
          </p>
        </span>
         
         
        <p>
          Also, there is a 
           
          <a
            href="/license"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            License
          </a>
           for the code 
          and it's probably a good idea to read the
           
          <a
            href="/privacy-policy"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Privacy Policy
          </a>
          .
        </p>
      </div>
    </footer>
  </div>
</div>
`;
